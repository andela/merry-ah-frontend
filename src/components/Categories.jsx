import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import painting from '../assets/images/painting.jpg';
import sketch from '../assets/images/sketch.jpg';
import mosaic from '../assets/images/mosaic.jpg';
import sculpture from '../assets/images/welcome.jpeg';
import architecture from '../assets/images/architecture.jpg';
import poetry from '../assets/images/poetry.jpg';

/**
 * @param {function} event
 *  @returns {JSX} jsx
 */
export default class Categories extends Component {
  /**
   * @param {object} props
   *  @returns {JSX} jsx
   */
  constructor(props) {
    super(props);
    this.state = {
      categoryList: [
        {
          objectID: 1,
          url: painting,
          title: 'PAINTING',
          category: 'painting',
        },
        {
          objectID: 2,
          url: sketch,
          title: 'FREEHAND SKETCH',
          category: 'freehand_sketch',
        },
        {
          objectID: 3,
          url: mosaic,
          title: 'MOSAIC',
          category: 'mosaic',
        },
        {
          objectID: 4,
          url: sculpture,
          title: 'SCULPTURE',
          category: 'sculpture',
        },
        {
          objectID: 5,
          url: architecture,
          title: 'ARCHITECTURE',
          category: 'architecture',
        },
        {
          objectID: 6,
          url: poetry,
          title: 'POETRY',
          category: 'poetry',
        },
      ],
    };
  }

  /**
     * @param {function} render
     *  @returns {JSX} jsx
     */
  render() {
    return (
      <section id="categories">
        <div>
          {
            this.state.categoryList.map(
              item => <div id="category" className="image"
                key={item.objectID}
                style={
                  {
                    background: `linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(${item.url}) no-repeat center `,
                    backgroundSize: 'cover',
                  }
                }>
                <div className="overlay">
                  <span>
                    <div id="category_link">
                      <Link to={`/art/${item.category}`}>{item.title}</Link>
                    </div>
                  </span>
                </div>
              </div>,
            )
          }
        </div>
      </section>
    );
  }
}
