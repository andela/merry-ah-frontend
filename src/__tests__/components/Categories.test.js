import React from 'react';
import { shallow } from 'enzyme';
import Categories from '../../components/Categories';

let wrapper;
beforeEach(() => {
 wrapper = shallow(<Categories />);
})
describe('Categories Component', () => {
  it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  })
})
