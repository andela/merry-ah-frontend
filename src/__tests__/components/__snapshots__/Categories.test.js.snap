// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component should match snapshot 1`] = `
<section
  id="categories"
>
  <div>
    <div
      className="image"
      id="category"
      key="1"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/painting"
            >
              PAINTING
            </Link>
          </div>
        </span>
      </div>
    </div>
    <div
      className="image"
      id="category"
      key="2"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/freehand_sketch"
            >
              FREEHAND SKETCH
            </Link>
          </div>
        </span>
      </div>
    </div>
    <div
      className="image"
      id="category"
      key="3"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/mosaic"
            >
              MOSAIC
            </Link>
          </div>
        </span>
      </div>
    </div>
    <div
      className="image"
      id="category"
      key="4"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/sculpture"
            >
              SCULPTURE
            </Link>
          </div>
        </span>
      </div>
    </div>
    <div
      className="image"
      id="category"
      key="5"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/architecture"
            >
              ARCHITECTURE
            </Link>
          </div>
        </span>
      </div>
    </div>
    <div
      className="image"
      id="category"
      key="6"
      style={
        Object {
          "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                            rgba(0, 0, 0, 0.5)),
                            url(test-file-stub) no-repeat center ",
          "backgroundSize": "cover",
        }
      }
    >
      <div
        className="overlay"
      >
        <span>
          <div
            id="category_link"
          >
            <Link
              replace={false}
              to="/art/poetry"
            >
              POETRY
            </Link>
          </div>
        </span>
      </div>
    </div>
  </div>
</section>
`;
